src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -1327,6 +1327,27 @@ public void Add(IEnumerable<T> items)
         }
 
         /// <summary>
+        /// Special add for TypeTable type entries that removes redundant file entries.
+        /// </summary>
+        internal void AddTypeTableTypesInfo(IEnumerable<T> items)
+        {
+            if (!(items is IEnumerable<SessionStateTypeEntry>)) { throw new PSInvalidOperationException(); }
+
+            lock (_syncObject)
+            {
+                foreach (var element in items)
+                {
+                    var typeEntry = element as SessionStateTypeEntry;
```

Comment:
Your check at the start of the function means you can cast instead of using `as`, which you should prefer because you're not testing if `typeEntry` is null. And I suppose we don't allow adding `null` entries?